client/components/modal.jsx

```diff
@@ -0,0 +1,111 @@
+import React from 'react';
+import Modal from 'react-bootstrap/Modal';
+import Button from 'react-bootstrap/Button';
+import AppContext from '../lib/app-context';
+
+export default class PostModal extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      postPicture: 'placeholder.png',
+      location: '',
+      caption: ''
+    };
+    this.fileInputRef = React.createRef();
+    this.handleChange = this.handleChange.bind(this);
+    this.handleCreatePost = this.handleCreatePost.bind(this);
+  }
+
+  handleChange(event) {
+    if (event.target.getAttribute('type') === 'file') {
+      this.setState({
+        postPicture: URL.createObjectURL(event.target.files[0])
+      });
+    } else {
+      const { name, value } = event.target;
+      this.setState({ [name]: value });
+    }
+  }
+
+  handleCreatePost(event) {
+    event.preventDefault();
+    const userId = this.context.user.userId;
+    const formData = new FormData();
+    formData.append('image', this.fileInputRef.current.files[0]);
+    formData.append('location', this.state.location);
+    formData.append('caption', this.state.caption);
+    formData.append('userId', userId);
+    fetch('/api/posts', {
+      method: 'POST',
+      body: formData
+    })
+      .then(res => {
+        res.json();
```

Comment:
If you're not going to to anything with the response body, you can skip parsing it.